Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM registry.redhat.io/ubi8:latest as builder
+
+# Install required RPMs and ensure that the packages were installed
+USER root
+RUN yum --setopt=tsflags=nodocs install -y unzip tar rsync shadow-utils \
+    && rpm -q unzip tar rsync shadow-utils
+
+
+# Install required RPMs and ensure that the packages were installed
+USER root
+RUN yum --setopt=tsflags=nodocs install -y java-11-openjdk-devel \
+    && rpm -q java-11-openjdk-devel
+
+# Install required RPMs and ensure that the packages were installed
+USER root
+RUN yum --setopt=tsflags=nodocs install -y maven \
+    && rpm -q maven
+
+
+RUN mkdir /target
+COPY . /approval-pam/src/
+WORKDIR /approval-pam/src/
+RUN mvn package -DoutputDirectory=/target
+
```

Comment:
I would suggest re-ordering these a bit, that way the directory creation happens before the copy. This will make the build faster due to being able to cache it. 

Also don't need to specify the `-D` option if we're just going to copy the file out of `$WORKDIR/target`

```suggestion
WORKDIR /approval-pam/src

COPY . .
RUN mvn package
```